src/main/java/com/myhome/controllers/HouseController.java

```diff
@@ -156,4 +158,36 @@ public HouseController(CommunityHouseRepository communityHouseRepository,
       return ResponseEntity.status(HttpStatus.NOT_FOUND).build();
     }
   }
+  @Operation(description = "Post Date Interval for houseId and Member Id")
+  @PostMapping(
+          path = "/house/member/interval",
+          produces = {MediaType.APPLICATION_JSON_VALUE, MediaType.APPLICATION_XML_VALUE},
+          consumes = {MediaType.APPLICATION_JSON_VALUE, MediaType.APPLICATION_XML_VALUE}
+  )
+  public ResponseEntity<HouseHistory> postInterval(@Valid @RequestBody HouseHistoryDto houseHistoryDto) {
+    log.trace("Received request to postinterval reuqest "+ CommonHelper.getStringFromObject(houseHistoryDto));
+
+    HouseHistory houseHistory = houseService.addInterval(houseHistoryDto);
+    if(!CommonHelper.empty(houseHistory) ){
+      return ResponseEntity.status(HttpStatus.CREATED).body(houseHistory);
+    }
+
+    return ResponseEntity.status(HttpStatus.BAD_REQUEST).body(houseHistory);
+  }
+
+  @Operation(description = "List History of House/Member")
+  @GetMapping(
+          path = "/house/history/{houseId}",
+          produces = {MediaType.APPLICATION_JSON_VALUE, MediaType.APPLICATION_XML_VALUE}
+  )
+  public ResponseEntity<List<HouseHistoryDto>> getHouseHistory(@PathVariable(required = true) String houseId,
```

Comment:
In general, it is better to wrap `List<HouseHistoryDto>` in a separate class(say `GetHouseHistoryResponse.java`) and return that instead.

This would allow us to easily extend the response to include any additional data in future.